### PR TITLE
[PT FE]: support aten::index

### PR DESCRIPTION
**Pull request by eaidova**
### Details:
 - *support aten::index* 
 transformation tested on swinv2, maxvit, ssd300 from torchvision, coatnets from timm

### Tickets:
 - *94303*

### CODE REVIEW

**Review by mvafin — CHANGES_REQUESTED** [review]

File: src/frontends/pytorch/src/transforms/aten_index_replacer.cpp [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+                return true;
+            }
+            auto adv_idx_count = advanced_ids.size();
+            auto rank = input_node->get_input_partial_shape(0).rank();
+            if (rank.is_dynamic()) {
+                FRONT_END_CHECK_IMPLEMENTED(false, "indexing for tensor with dynamic rank is not implemented ");
+            }
+            auto input_shape = std::make_shared<v3::ShapeOf>(input_node);
+            auto zero = v0::Constant::create(element::i64, Shape{}, {0});
+            auto input_dims = std::make_shared<v1::Split>(input_shape, zero, rank.get_length());
```
Comment (mmikolajcz) [inline src/frontends/pytorch/src/transforms/aten_index_replacer.cpp:158]: While testing it on Cityscapes/mask_rcnn_R_50_FPN models we saw that there might be an issue with Split op when splitting dimension of 3 into 2 splits due to OV limitations.

Comment (mvafin) [reply src/frontends/pytorch/src/transforms/aten_index_replacer.cpp:158]: ov `Split` only works for the cases when dimensions are divisible, like in this case where we split shape by rank

File: src/frontends/pytorch/src/transforms/aten_index_replacer.cpp [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+            //       t: [x_1, x_2, ..., x_m, y_1, y_2, ..., y_n]
+            // and use gather
+            //       t: [x_1 * x_2 * ... * x_m, y_1 * y_2 * ... * y_n]
+            //       tensor index = \sum_{i=1}^m (ind_i * \prod_{j=i+1}^m (x_j))
+            // After gather, reshape and transpose back.
+            auto ids = list_indicies->input_values();
+            std::vector<size_t> advanced_ids;
+            std::vector<bool> is_masked_bool;
+            OutputVector masked_indicies;
+            // for case when index is bool e.g. x[x>0], replace index with non_zero
```
[the remark this answers is not in the document]
Comment (eaidova) [reply src/frontends/pytorch/src/transforms/aten_index_replacer.cpp:102]: x>0 is elementwise condition, this translated before index, index just got boolean mask [True, False, ...] as result x > 0 for all elements to get indexes we need to get all elements where condition != False, that is why nonzero used

File: src/frontends/pytorch/src/transforms/aten_index_replacer.cpp [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+    std::shared_ptr<Node> output_shape;
+    if (axis == 0) {
+        output_shape = v0::Constant::create(element::i64, Shape{2}, {1, -1});
+    } else if (axis == 1) {
+        output_shape = v0::Constant::create(element::i64, Shape{2}, {0, -1});
+    } else {
+        const auto value_shape = std::make_shared<v3::ShapeOf>(value);
+        const auto value_rank = std::make_shared<v3::ShapeOf>(value_shape);
+        const auto axis_node = v0::Constant::create(element::i64, Shape{}, {axis});
+        const auto first_part_dims = std::make_shared<v1::StridedSlice>(value_shape,
```
[the remark this answers is not in the document]
Comment (eaidova) [reply src/frontends/pytorch/src/transforms/aten_index_replacer.cpp]: thanks, changed

File: src/frontends/pytorch/src/transforms/aten_index_replacer.cpp [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+                                                                        v0::Constant::create(element::i64, {1}, {0}),
```
[the remark this answers is not in the document]
Comment (eaidova) [reply src/frontends/pytorch/src/transforms/aten_index_replacer.cpp]: separated

**Review by mvafin — APPROVED** [review]

**Pull request merged**